python/lsst/analysis/tools/actions/scalar/scalarActions.py

```diff
@@ -444,3 +446,39 @@ def __call__(self, data: KeyedData, **kwargs) -> Scalar:
                 return value
         else:
             return scalarA / scalarB
+
+
+class MagPercentileAction(ScalarFromVectorAction):
+    """Calculates the magnitude at the given percentile for completeness"""
+
+    matchDistanceKey = Field[str]("Match distance Vector")
+    fluxUnits = Field[str](doc="Units for the column.", default="nanojansky")
+    percentile = Field[float](doc="The percentile to find the magnitude at.", default=50.0, check=isPercent)
+
+    def getInputSchema(self) -> KeyedDataSchema:
+        return (
+            (self.matchDistanceKey, Vector),
+            (self.vectorKey, Vector),
+        )
+
+    def __call__(self, data: KeyedData, **kwargs) -> Scalar:
+        matched = np.isfinite(data[self.matchDistanceKey])
+        fluxValues = data[self.vectorKey.format(**kwargs)]
+        values = fluxToMag(fluxValues, flux_unit=u.Unit(self.fluxUnits))
+        nInput, bins = np.histogram(
+            values,
+            range=(np.nanmin(values), np.nanmax(values)),
+            bins=100,
+        )
+        nOutput, _ = np.histogram(
+            values[matched],
+            range=(np.nanmin(values[matched]), np.nanmax(values[matched])),
+            bins=bins,
+        )
+        # Find bin where the fraction recovered first falls below 0.5
```

Comment:
I think this comment is left over from before.

Reply:
The whole action is.